flow.py

```diff
@@ -179,6 +183,7 @@ def receive_packet(self, packet):
         if isinstance(packet, DataPacket):
             self.acknowledge(packet)
         elif isinstance(packet, AcknowledgementPacket):
+            self.ns.record_packet_ack_time(self.flow_id, packet.packet_id)
```

Comment:
This shouldn't be here, since it's possible that the packet expired/timed out. Fixing now.